### DISCUSSION TITLE
Regarding the problem that it cannot use the hot update function in Vue.js

### DISCUSSION THREAD

**Discussion by shuyingegit** [coder/code-server#5855] [Q&A]
In Vue In the development of js, the following requests are designed to update the page in time when modifying the source code.

eg. {URL}/sockjs-node/info? t=1670553943109

However, when we use [/proxy/(: proxy)] to access the development page of vue.js, the hot deployment function of sockjs cannot be realized through the following configurations

Deployment method 1: sockPath: "sockjs node"
Deployment method 2: sockPath: "/proxy/9527/sockjs node/"

**Comment by jsjoeio** [comment]
This is old, but have you tried this? https://github.com/coder/code-server/discussions/3912#discussioncomment-1140449

**Reply by shuyingegit** [reply]
Thank you for your answer.
I tried the above method, but it didn't work.
The configuration here is also different from the above scenario.

I mean that the code server is configured under the subPath of nginx, so the code server service is opened with the following URL.
eg：https://ip:port/subPath/

In this case, whether you use proxy or absproxy to open the development page of vue.js, you can also open it. Examples of urls are as follows
eg：
      https://ip:port/subPath/proxy/(:port)
      https://ip:port/subPath/absproxy/(:port)

I have made a corresponding attempt for the above example, but I can always open the development page, but I can't implement hot deployment when modifying code, and the page can't get the latest status.

**Reply by shuyingegit** [reply]
ps: error image

subPath: 210

![image](https://user-images.githubusercontent.com/27947525/206976492-0352e58f-d522-433b-8d71-c5249c48103a.png)

**Reply by jsjoeio** [reply]
Hmm 🤔 Part of me feels like you may have to add something particular to your nginx configuration but I'm not 100% sure. 

Possibly related? https://github.com/vitejs/vite/issues/6102

**Reply by shuyingegit** [reply]
My nginx configuration is as follows：
``` 
server {
  server_name xxx.com;
  listen 4 ssl http2;
  ssl_certificate /home/aaa/.acme.sh/*.xxx.com/fullchain.cer;
  ssl_certificate_key /home/aaa/.acme.sh/*.xxx.com/*.xxx.com.key;
  ssl_protocols TLSv1 TLSv1.1 TLSv1.2 TLSv1.3;
  if ($scheme = http) {
    return 301 https://$host:4$request_uri;
  }
  proxy_set_header X-Real-IP $remote_addr;
  proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
  proxy_set_header X-NginX-Proxy true;
  underscores_in_headers on;
  location /210/ {
    # 210 - CS;
    proxy_pass http://10.10.10.210:9999/;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $http_host;
    proxy_set_header X-Forwarded-Port $server_port;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_redirect http:// https://;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-NginX-Proxy true;
  }
}

```

Add the following configuration. Although the sockjs request can be successfully forwarded, this is contrary to the logic of the code server automatic proxy.
The purpose of code server is to automatically forward service ports such as 9527 or 9528 to users for use, so that users no longer pay attention to the manual forwarding process of service ports.
If you must manually forward the sockjs request on port 9527 through nginx, it means that the service on port 9527 can also be directly completed using nginx forwarding, instead of the code server proxy automatic proxy program.
```
  location /210/proxy/9527/sockjs-node/ {
    proxy_pass http://10.10.10.210:9527/sockjs-node/;
    proxy_set_header Host $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Host $http_host;
    proxy_set_header X-Forwarded-Port $server_port;
    proxy_set_header X-Forwarded-Proto $scheme;
    proxy_http_version 1.1;
    proxy_set_header Upgrade $http_upgrade;
    proxy_set_header Connection "upgrade";
    proxy_redirect http:// https://;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-NginX-Proxy true;
  }
```

**Reply by jsjoeio** [reply]
I wish I could help more but I'm not totally sure where we'd need to dig in. Happy to test things locally or debug here with you!
